plugins/services/src/js/service-configuration/ServiceConfigDisplay.js

```diff
@@ -0,0 +1,128 @@
+import React from 'react';
+
+import ConfigurationMap from '../../../../../src/js/components/ConfigurationMap';
+import ConfigurationMapHeading from '../../../../../src/js/components/ConfigurationMapHeading';
+import ConfigurationMapLabel from '../../../../../src/js/components/ConfigurationMapLabel';
+import ConfigurationMapRow from '../../../../../src/js/components/ConfigurationMapRow';
+import ConfigurationMapSection from '../../../../../src/js/components/ConfigurationMapSection';
+import ConfigurationMapValue from '../../../../../src/js/components/ConfigurationMapValue';
+import GeneralServiceConfigSection from './GeneralServiceConfigSection';
+import NetworkingServiceConfigSection from './NetworkingServiceConfigSection';
+import StorageServiceConfigSection from './StorageServiceConfigSection';
+import EnvironmentVariablesServiceConfigSection from './EnvironmentVariablesServiceConfigSection';
+import LabelsServiceConfigSection from './LabelsServiceConfigSection';
+import HealthChecksServiceConfigSection from './HealthChecksServiceConfigSection';
+import ServiceConfigDisplayUtil from '../utils/ServiceConfigDisplayUtil';
+import Util from '../../../../../src/js/utils/Util';
+
+const serviceConfigDisplayList = [
+  GeneralServiceConfigSection,
+  NetworkingServiceConfigSection,
+  StorageServiceConfigSection,
+  EnvironmentVariablesServiceConfigSection,
+  LabelsServiceConfigSection,
+  HealthChecksServiceConfigSection
+];
+
+const shouldExcludeItem = (row, appConfig) => {
+  switch (row.key) {
+    case 'gpus':
+      return !Util.findNestedPropertyInObject(appConfig, 'gpus');
+    case 'container.volumes':
+      return !Util.findNestedPropertyInObject(
+        appConfig, 'container.volumes.length'
+      );
+    case 'healthChecks':
+      return !Util.findNestedPropertyInObject(appConfig, 'healthChecks.length');
+    case 'secrets':
+      return appConfig.secrets == null
+        || Object.keys(appConfig.secrets).length === 0;
+    case 'env':
+      return appConfig.env == null
+        || Object.keys(appConfig.env).length === 0;
+    case 'labels':
+      return appConfig.labels == null
+        || Object.keys(appConfig.labels).length === 0;
+    default:
+      return false;
+  }
+};
+
+const getDisplayValue = (type, value) => {
+  // If the row's type is pre, we wrap it in a pre tag.
+  if (type === 'pre') {
+    return <pre className="flush transparent wrap">{value}</pre>;
+  }
+
+  return ServiceConfigDisplayUtil.getDisplayValue(value);
+};
+
+const ServiceConfigDisplay = ({appConfig}) => {
+  const configurationMapSections = serviceConfigDisplayList.map(
+    (section, sectionIndex) => {
+      let configurationMapRows = section.values.filter((row) => {
+        // Some rows must be excluded if relevant data is missing.
```

Comment:
I think it's fine to not show a row or section if there is nothing configured, or no default e.g. if no volume configured then don't show volumes section. You're right that there could also be an advantage showing it's not configured, but I would start with hiding it then based on feedback decide if we actually want to show everything.